File: src/main/scala/com/funapps/weather/models/WeatherResponse.scala
package com.funapps.weather.models

import cats.effect.Concurrent
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}
import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.circe.{jsonEncoderOf, jsonOf}

case class WeatherResponse(
  weather: Vector[WeatherSummary],
  main: BasicWeatherStats
)

object WeatherResponse {
  implicit val decoder = deriveDecoder[WeatherResponse]
  implicit val encoder = deriveEncoder[WeatherResponse]
  implicit def entityDecoder[F[_]: Concurrent]: EntityDecoder[F, WeatherResponse] = jsonOf
  implicit def entityEncoder[F[_]]: EntityEncoder[F, WeatherResponse] = jsonEncoderOf
}

File: python_backend/app/internal/model/weather_response.py
from pydantic import BaseModel

from .coordinates import Coordinates
from .weather_summary import WeatherSummary
from .basic_weather_stats import BasicWeatherStats
from .wind_data import WindData

class WeatherResponse(BaseModel):
    coord: Coordinates
    weather: list[WeatherSummary]
    main: BasicWeatherStats
    wind: WindData
